### DISCUSSION TITLE
Waste polygons (straight lines in places where they should not appear)

### DISCUSSION THREAD

**Discussion by ArnauGomar** [NetTopologySuite/NetTopologySuite#561] [Q&A]
Hello everybody. I am working with this wonderful library for my final degree project and I am running into an error that I do not know how to solve. When executing multiple intersections, straight lines appear in places that should not appear. Anyone have any idea what I may be doing wrong? Thank you!

Capture1: https://gyazo.com/6296208fd1bcb98a4c43017b04910d21
Capture2: https://gyazo.com/e2e5afc65519754fd54812a653643771

I am intersecting different polygons and sometimes these lines appear to me as a result in areas where there should be nothing (I also use the difference between polygons). 

If someone could tell me what I am doing wrong I would greatly appreciate it. Thank you so much for your time! :)

**Comment by FObermaier** [comment]
Are those lines common boundaries to the intersecting polygons?

**Reply by ArnauGomar** [reply]
The lines that appear are related to the original polygons (fig 1 and 2), but I think the problem is in the execution of a difference and not in the calculation of the intersection.

fig 1: https://gyazo.com/b3cd274201f92253159c5d48c2722582 (Wrong result)
fig 2: https://gyazo.com/4e2abc336cee5b0b95e460c176b43a0e (Wrong result + original coverages)

I am developing a software that calculates the overlap of coverages, I look for the intersections between different polygons in order to determine if in an area 1, 2 or more coverings act.

In order to do this, I first calculate all the intersections and then differentiate to get the end result. I classify the intersections by level, level 1 only one cover acts, level 2 two covers, level 3 ... Once I have all the intersections I join them by levels and execute the difference with all the intersections of lower levels in order to achieve the correct result.

The case I show you below, you can see an intersection of 3 coverages (pink polygon and therefore level 3, fig 3) if only these 3 coverages the result would be what you see, the problem is that it is a case where 9 coverages act, therefore, the pink polygon is not the end result.

fig 3: https://gyazo.com/3d0a0cbf00e7fbd03e37fa5430678686 (Intersect lvl 3)
fig 4: https://gyazo.com/29f886488ce53a3939d1c6978a4f50db (End result, lvl 4, 5 and 6)

Precisely in this particular case, there are coverages of level 4, 5 and 6 (orange, fig 4) that overlap on the pink polygon (level 3), therefore to obtain a correct result a difference has been executed (set of polygons of level 4, orange, on the pink polygon of level 3) and its theoretical result should be an “empty polygon” but it is not the case, these straight lines appear.

If you look closely, the lines follow a stretch of the border of one of the original covers (fig 1 and 2). I'm also attaching a KML with level 4, 5 and 6 intersections, the original level 3 and finally the result, which according to my way of understanding the algorithm, is wrong.

[Errors.zip](https://github.com/NetTopologySuite/NetTopologySuite/files/7318348/Errors.zip)

Thank you very much for your attention, 

Arnau Gomar

**Reply by FObermaier** [reply]
Sorry for the delay. I'd say these artefacts are not due to wrong algorithms but slight mismatches in coordinates due to parsing/digitizing/... issues. I'd say you can safely ignore theses artifacts as these polygons almost collapse (`Geometry.Area` is very very small).

The spikes you are getting with some of the geometries can be removed by 
```C#
var geom = NetTopologySuite.Operations.Overlay.GeometrySnapper.SnapToSelf(geomWithSpikes, snapTolerance, true);
```
